### PR TITLE
gh-96414: Inline code examples in sqlite3 docs

### PR DESCRIPTION
**Pull request by erlend-aasland**
This PR inlines all sqlite3 doc examples into `Doc/library/sqlite3.rst`.
Except from adjusting the SQL style, the examples are mostly kept as they are;
some of them work well, others can be improved in later PRs.

* Issue: gh-96414

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@CAM-Gerlach, do you think this is worth it? IMO, the benefits of inlining the examples (being able to doctest them, etc.) outweigh the churn. I don't expect you to review each line here; I've just moved a lot of lines :)

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -738,7 +784,16 @@ Connection objects
...
 
-      .. literalinclude:: ../includes/sqlite3/md5func.py
+      .. doctest::
+
+         >>> import hashlib
+         >>> def md5sum(t):
+         ...     return hashlib.md5(t).hexdigest()
+         >>> con = sqlite3.connect(":memory:")
+         >>> con.create_function("md5", 1, md5sum)
+         >>> for row in con.execute("select md5(?)", (b"foo",)):
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]: Wouldn't this be

```suggestion
         >>> for row in con.execute("SELECT md5(?)", (b"foo",)):
```

Or no?

File: Doc/library/sqlite3.rst [hunk 2]
```diff
@@ -925,7 +1077,43 @@ Connection objects
...
+
+         # example from SQLite wiki
+         con.execute("create virtual table recipe using fts3(name, ingredients)")
+         con.executescript("""
+             insert into recipe (name, ingredients) values ('broccoli stew', 'broccoli peppers cheese tomatoes');
+             insert into recipe (name, ingredients) values ('pumpkin stew', 'pumpkin onions garlic celery');
+             insert into recipe (name, ingredients) values ('broccoli pie', 'broccoli cheese onions flour');
+             insert into recipe (name, ingredients) values ('pumpkin pie', 'pumpkin sugar flour butter');
+             """)
+         for row in con.execute("select rowid, name, ingredients from recipe where name match 'pie'"):
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
         con.execute("CREATE VIRTUAL TABLE recipe USING fts3(name, ingredients)")
         con.executescript("""
             INSERT INTO recipe (name, ingredients) VALUES ('broccoli stew', 'broccoli peppers cheese tomatoes');
             INSERT INTO recipe (name, ingredients) VALUES ('pumpkin stew', 'pumpkin onions garlic celery');
             INSERT INTO recipe (name, ingredients) VALUES ('broccoli pie', 'broccoli cheese onions flour');
             INSERT INTO recipe (name, ingredients) VALUES ('pumpkin pie', 'pumpkin sugar flour butter');
             """)
         for row in con.execute("SELECT rowid, name, ingredients FROM recipe WHERE name MATCH 'pie'"):
```

Code style

File: Doc/library/sqlite3.rst [hunk 3]
```diff
@@ -1712,7 +1946,26 @@ This can be implemented by adding a ``__conform__(self, protocol)``
...
+           self.x, self.y = x, y
+
+       def __conform__(self, protocol):
+           if protocol is sqlite3.PrepareProtocol:
+               return f"{self.x};{self.y}"
+
+   con = sqlite3.connect(":memory:")
+   cur = con.cursor()
+
+   cur.execute("select ?", (Point(4.0, -3.2),))
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
   cur.execute("SELECT ?", (Point(4.0, -3.2),))
```

File: Doc/library/sqlite3.rst [hunk 4]
```diff
@@ -1722,7 +1975,27 @@ The other possibility is to create a function that converts the Python object
...
+
+   def adapt_point(point):
+       return f"{point.x};{point.y}"
+
+   sqlite3.register_adapter(Point, adapt_point)
+
+   con = sqlite3.connect(":memory:")
+   cur = con.cursor()
+
+   cur.execute("select ?", (Point(1.0, 2.5),))
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
   cur.execute("SELECT ?", (Point(1.0, 2.5),))
```

**Review by CAM-Gerlach — COMMENTED** [review]
IMO, Diataxis stresses that examples given must work and work reliably, especially for tutorials. So, given this just requires moving them around, actually testing them would seem to be worth it. 

However, just to make sure, is the doctest CI check actually correctly failing if a doctest fails? Perhaps I'm mistaken, but at least per my recollection of a discussion on the Docs Community Discord, doctests failing doesn't actually fail the build, it just produces messages/output—but perhaps I'm entirely mistaken on that, and its just referring to warnings or something.

I did have some trivial comments regarding consistent SQL code style, given you mentioned that conforming that was in scope for this PR.

**Comment by erlend-aasland** [comment]
> However, just to make sure, is the doctest CI check actually correctly failing if a doctest fails?

Yes. I learned that when I applied doctests to the tutorial. Several of the examples had inconsistencies that made the code and thus the doctest fail.

**Comment by erlend-aasland** [comment]
> I did have some trivial comments regarding consistent SQL code style, given you mentioned that conforming that was in scope for this PR.

Thanks! :)

**Pull request merged**